### PR TITLE
Support quant procedure on XPU

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by thuang6** [comment]
My question is that if the procedure is on GPU layer-wise quant behavior does not apply to GPU memory, right? in that case, GPU memory need to be able to hold full weights?

**Comment by Kaihui-intel** [comment]
> My question is that if the procedure is on GPU layer-wise quant behavior does not apply to GPU memory, right? in that case, GPU memory need to be able to hold full weights?

Actually, the packing behavior still takes place on the CPU, but the GPU memory does not need to hold all of the original weights.  It only needs to hold the quantized weights and the weights that have not been quantized.

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/transformers/weight_only/text-generation/README.md [hunk 1]
```diff
@@ -111,7 +113,7 @@ Intel-extension-for-pytorch dependencies are in oneapi package, before install i
 ### Create Environment​
 Pytorch and Intel-extension-for-pytorch version for intel GPU > 2.1 are required, python version requests equal or higher than 3.9 due to [text evaluation library](https://github.com/EleutherAI/lm-evaluation-harness/tree/master) limitation, the dependent packages are listed in requirements_GPU.txt, we recommend create environment as the following steps. For Intel-exension-for-pytorch, we should install from source code now, and Intel-extension-for-pytorch will add weight-only quantization in the next version.
 
->**Note**: please install transformers==4.40.2.
+>**Note**: please install transformers==4.38.1.
```
Comment (thuang6) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/transformers/weight_only/text-generation/README.md:116]: is this transformers version higher enough to support new models like Llama3.1-8B?  I remember it has requirements on transformers version

**Review by thuang6 — APPROVED** [review]

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/transformers/weight_only/text-generation/README.md (hunk 1 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/transformers/weight_only/text-generation/README.md:116]: We currently support Llama 3-8B, not Llama 3.1-8B.

**Review by changwangss — APPROVED** [review]

**Review by XuehaoSun — APPROVED** [review]

**Pull request merged**